build.sbt

```diff
@@ -23,39 +23,40 @@ lazy val core = mkModule("core")
   .settings(
     libraryDependencies ++= List(
       // TODO: replace Calvin's version with the canonical one once its released
-      "com.github.kaizen-solutions.trace4cats-newrelic" %% "trace4cats-newrelic-http-exporter" % "0.12.2",
+      trace4cats("newrelic-http-exporter"),
       "org.http4s" %% "http4s-async-http-client" % "0.23.10",
     )
   )
 
 lazy val sttp = mkModule("sttp")
   .settings(
     libraryDependencies ++= List(
-      "com.softwaremill.sttp.client3" %% "async-http-client-backend-zio1" % "3.6.1",
+      "com.softwaremill.sttp.client3" %% "async-http-client-backend-zio" % "3.5.2",
       trace4cats("sttp-client3"),
-    )
+    ).map { _.excludeAll(ExclusionRule("org.scala-lang.modules")) }
```

Comment:
So we ran into a problem on another scala 3 project where sttp dependencies had a transitive dependency on scala-collection-compat: this in turn gave us surprising compilation issues in a scala 3 project.

I've excluded here because, well, copy/paste. If I remove the exclusion rule, everything compiles ok, so I guess I can restore if you think that best: in theory, we shouldn't need that dependency because we are not cross compiling for anything less than scala 2.13 here.

Let me know what you think

Reply:
hmm maybe add a comment like "if you encounter this compile error, add this exclusion rule"?